src/nvim/event/process.c

```diff
@@ -21,9 +21,12 @@
 # include "event/process.c.generated.h"
 #endif
 
-// Time (ns) for a process to exit cleanly before we send TERM/KILL.
-#define TERM_TIMEOUT 1000000000
-#define KILL_TIMEOUT (TERM_TIMEOUT * 2)
+// Time for a process to exit cleanly before we send KILL.
+#define KILL_TIMEOUT_MS 2000
+#define KILL_TIMEOUT_NS (KILL_TIMEOUT_MS * 1000000)
+// Time for a pty process to exit due to SIGHUP before we send TERM.
+#define PTY_TERM_TIMEOUT_MS 1000
+#define PTY_TERM_TIMEOUT_NS (PTY_TERM_TIMEOUT_MS * 1000000)
```

Comment:
SIGHUP is used with pty processes by default (without timeout), and SIGTERM with others (without timeout).

If SIGHUP is not enough (as I understood from @jamessan), then SIGTERM gets used after 1s for pty processes (which could also be 2s though).  It should not send SIGKILL though directly, but only after another timeout.
I think having a single value for the timeout (of 2s) could simplify it a bit, which results in 4s for a non-responsive pty process.  Two different macros/values/names for it make it more descriptive though, and a shorter timeout for SIGTERM can be used then.